AppActionVisualizer/app/src/main/java/com/example/appactionvisualizer/ui/adapter/ActionRecyclerViewAdapter.java

```diff
@@ -0,0 +1,81 @@
+package com.example.appactionvisualizer.ui.adapter;
+
+import android.content.Context;
+import android.content.Intent;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.appactionvisualizer.R;
+import com.example.appactionvisualizer.constants.Constant;
+import com.example.appactionvisualizer.databean.Action;
+import com.example.appactionvisualizer.databean.AppAction;
+import com.example.appactionvisualizer.ui.activity.FulfillmentActivity;
+import com.example.appactionvisualizer.utils.Utils;
+
+/**
+ * Adapter of ActionActivity Recyclerview
+ */
+public class ActionRecyclerViewAdapter extends RecyclerView.Adapter<ActionRecyclerViewAdapter.ViewHolder> {
+
+  private AppAction appAction;
+  private Context context;
+
+  public ActionRecyclerViewAdapter(AppAction items, Context context) {
+    this.context = context;
+    appAction = items;
+  }
+
+  @Override
+  public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+    View view = LayoutInflater.from(context)
+        .inflate(R.layout.action_rv_item, parent, false);
+    return new ViewHolder(view);
+  }
+
+  @Override
+  public void onBindViewHolder(final ViewHolder holder, final int position) {
+    final Action action = appAction.getActions().get(position);
+    holder.tvActionName.setText(action.getIntentName() + " (" + action.getFulfillmentArrayList().size() + ")" );
```

Comment:
we can use placeholder for action.getIntentName() + " (" + action.getFulfillmentArrayList().size() + ")", define the string in the XML file with placeholder

Reply:
Agree. I'll use placeholders.